src/app/public/modules/datepicker/datepicker.component.ts

```diff
@@ -1,67 +1,83 @@
 import {
+  ChangeDetectionStrategy,
   Component,
   EventEmitter,
-  ViewChild
+  OnDestroy,
+  ViewChild,
+  ChangeDetectorRef
 } from '@angular/core';
 
-import { Subject } from 'rxjs/Subject';
-
-import { SkyDatepickerCalendarComponent } from './datepicker-calendar.component';
-
 import {
-  SkyDropdownComponent,
   SkyDropdownMessage,
   SkyDropdownMessageType
 } from '@skyux/popovers';
 
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import {
+  SkyDatepickerCalendarComponent
+} from './datepicker-calendar.component';
+
 @Component({
   selector: 'sky-datepicker',
   templateUrl: './datepicker.component.html',
-  styleUrls: ['./datepicker.component.scss']
+  styleUrls: ['./datepicker.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class SkyDatepickerComponent {
-  @ViewChild(SkyDatepickerCalendarComponent)
-  public calendar: SkyDatepickerCalendarComponent;
+export class SkyDatepickerComponent implements OnDestroy {
+  public get disabled(): boolean {
+    return this._disabled;
+  }
+
+  public set disabled(value: boolean) {
+    this._disabled = value;
+    this.changeDetector.markForCheck();
+  }
+
+  public get dropdownController(): Observable<SkyDropdownMessage> {
+    return this._dropdownController;
+  }
 
-  @ViewChild(SkyDropdownComponent)
-  public dropdown: SkyDropdownComponent;
```

Comment:
Just checking - was removing this `public` property deliberate?

Reply:
It was. It's not being accessed anywhere within the `datetime` library.